pysteps/nowcasts/linear_blending.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+"""
+pysteps.nowcasts.linear_blending
+================================
+
+Implementation of the linear blending between nowcast and NWP data.
+
+.. autosummary::
+    :toctree: ../generated/
+
+    forecast
+"""
+
+import numpy as np
+from pysteps import nowcasts
+from pysteps.utils import transformation
+
+
+def forecast(
+    precip,
+    velocity,
+    timesteps,
+    timestep,
+    nowcast_method,
+    R_nwp=None,
+    start_blending=120,
+    end_blending=240,
+    nowcast_kwargs=dict(),
+):
+
+    """Generate a forecast by linearly blending nowcasts with NWP data
+
+    Parameters
+    ----------
+    precip: array_like
+      Array containing the input precipitation field(s) ordered by timestamp
+      from oldest to newest. The time steps between the inputs are assumed
+      to be regular.
+    velocity; array_like
+      Array of shape (2, m, n) containing the x- and y-components of the advection
+      field. The velocities are assumed to represent one time step between the
+      inputs. All values are required to be finite.
+    timesteps: int
+      Number of time steps to forecast.
+    timestep: int or float
+      The time difference (in minutes) between consecutive forecast fields.
+    nowcast_method: {'anvil', 'eulerian', 'lagrangian', 'extrapolation', 'lagrangian_probability', 'sprog', 'steps', 'sseps'}
+      Name of the nowcast method used to forecast the precipitation.
+    R_nwp: array_like or NoneType, optional
+      Array of shape (timesteps, m, n) in the case of no ensemble or
+      of shape (n_ens_members, timesteps, m, n) in the case of an ensemble
+      containing the NWP precipitation fields ordered by timestamp from oldest
+      to newest. The time steps between the inputs are assumed to be regular
+      (and identical to the time step between the nowcasts). If no NWP
+      data is given the value of R_nwp is None and no blending will be performed.
+    start_blending: int, optional
+      Time stamp (in minutes) after which the blending should start. Before this
+      only the nowcast data is used.
+    end_blending: int, optional
+      Time stamp (in minutes) after which the blending should end. Between
+      start_blending and end_blending the nowcasts and NWP data are linearly
+      merged with each other. After end_blending only the NWP data is used.
+    nowcast_kwargs: dict, optional
+      Dictionary containing keyword arguments for the nowcast method.
+
+    Returns
+    -------
+    R_blended: ndarray
+      Array of shape (timesteps, m, n) in the case of no ensemble or
+      of shape (n_ens_members, timesteps, m, n) in the case of an ensemble
```

Comment:
With the way it is implemented at this moment: n_ens_members equals the maximum number of ensemble members in the input, so either the number of the nowcast or that of the NWP. Perhaps good to add that here.

Reply:
(B.t.w., I can add that too, but I'll wait for the responses - and thus possible changes - of the others)